clients/web/src/app.js

```diff
@@ -59,14 +99,31 @@ girder.App = girder.View.extend({
      * Changes the current body view to the view class specified by view.
      * @param view The view to display in the body.
      * @param [settings={}] Settings to pass to the view initialize() method.
+     * @param opts Additional options for navigateTo, if any.
      */
-    navigateTo: function (view, settings) {
+    navigateTo: function (view, settings, opts) {
         var container = this.$('#g-app-body-container');
 
         this.globalNavView.deactivateAll();
 
         settings = settings || {};
 
+        if (opts && opts.layout) {
+            if (girder.layout !== opts.layout) {
+                if (_.has(this.layoutRenderMap, opts.layout)) {
+                    // set a layout if opts specifies one different from current
+                    girder.layout = opts.layout;
+                    this.layoutRenderMap[girder.layout]();
```

Comment:
Let's change these lines to `this.layoutRenderMap[girder.layout].call(this, opts);`. That will let us remove the `_this` bindings above and just use regular `this` instead, and also will allow any special options for layout rendering to be passed through to the function.